tests/unit/browser/webengine/test_webenginetab.py

```diff
@@ -244,3 +244,140 @@ def test_clipboard_value(self):
             pytest.skip("enum member not available")
         assert clipboard in permissions_cls._options
         assert clipboard in permissions_cls._messages
+
+
+class TestPageLifecycle:
+
+    @pytest.fixture
+    def set_state_spy(
+        self,
+        webengine_tab: webenginetab.WebEngineTab,
+        monkeypatch,
+        mocker,
+    ):
+        set_state_spy = mocker.Mock()
```

Comment:
Should this maybe be a [`mocker.spy()`](https://pytest-mock.readthedocs.io/en/latest/usage.html#spy) so that the original methods still get called (also potentially catching issues with that)?

Reply:
I think I used the wrong name to be honest. I don't want it to call the backend. So I guess it should just be called a mock, not a spy. As a unit test, I just want it to focus on the logic in our codebase. There is a end2end test to test that the page actually does what we expect. Calling the actual page method might not call any issues, but why complicate things?